File: EduLink.Domain/Entities/States/CanceladaState.cs
using EduLink.Domain.Enums;

namespace EduLink.Domain.Entities.States;

public class CanceladaState : ReservaState
{
    public override EstadoReserva Nombre => EstadoReserva.Cancelada;
    // No permitimos transaccíones ( o sea que los metodos por defautl de reservastate no se modifican)
}

File: app/domain/entities/states/cancelada_state.rb
require_relative './reserva_state'
require_relative '../../enums/estado_reserva' # Corrected path

module Entities
  module States
    class CanceladaState < ReservaState
      def nombre
        Enums::EstadoReserva::CANCELADA
      end
      # No permitimos modificaciones, punto muerto
    end
  end
end
